infra/src/main/kotlin/fi/fta/geoviite/infra/tracklayout/LayoutContextData.kt

```diff
@@ -0,0 +1,242 @@
+package fi.fta.geoviite.infra.tracklayout
+
+import com.fasterxml.jackson.annotation.JsonIgnore
+import fi.fta.geoviite.infra.common.*
+import fi.fta.geoviite.infra.common.DataType.STORED
+import fi.fta.geoviite.infra.common.DataType.TEMP
+import fi.fta.geoviite.infra.logging.Loggable
+
+// TODO: GVT-2426 This enum is an outdated concept that won't serve well with designs. It's only used by the front. Can we refactor it out?
+enum class ContextType {
+    OFFICIAL, NEW_DRAFT, EDITED_DRAFT, DESIGN, NEW_DESIGN_DRAFT, EDITED_DESIGN_DRAFT
+}
+
+interface LayoutContextAware<T> {
+    val id: DomainId<T>
+    val dataType: DataType
+    // TODO: GVT-2426 if we keep this, the field should be renamed
+//    val contextType: ContextType
+    val draftType: ContextType
+    @get:JsonIgnore val rowId: DomainId<T>
+    @get:JsonIgnore val isDraft: Boolean
+    @get:JsonIgnore val isOfficial: Boolean
+    @get:JsonIgnore val isDesign: Boolean
+    @get:JsonIgnore val isEditedDraft: Boolean
+    @get:JsonIgnore val isNewDraft: Boolean
+}
+
+sealed class LayoutConcept<T : LayoutConcept<T>>(contextData: LayoutContextData<T>) :
+    LayoutContextAware<T> by contextData, Loggable {
+
+    abstract val version: RowVersion<T>?
+    abstract val contextData: LayoutContextData<T>
+}
+
+sealed class LayoutContextData<T> : LayoutContextAware<T> {
+    @get:JsonIgnore
+    override val isDraft: Boolean get() = this is MainDraftContextData || this is DesignDraftContextData
```

Comment:
Tein nämä jotta menisi pienemmällä muutoksella, mutta pääpointti on että nuo eri kontekstit on oikeasti eri tyypit ja kukin kantaa ne tiedot mitkä siihen tyyppiin kuuluu. Voisin katsoa näitä overridejä vielä uudelleen koska on vähän sekasikiö rakenne että noita joudutaan tarkastelemaan kantaluokassa.